pkg/yurthub/filter/podenvupdater/filter.go

```diff
@@ -0,0 +1,78 @@
+package podenvupdater
+
+import (
+	"github.com/openyurtio/openyurt/pkg/yurthub/filter"
+	"github.com/openyurtio/openyurt/pkg/yurthub/filter/base"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/sets"
+)
+
+const (
+	// FilterName filter is used to mutate the Kubernetes service host
+	// in order for pods on edge nodes to access kube-apiserver via Yurthub proxy
+	FilterName = "podenvupdater"
+
+	envVarServiceHost = "KUBERNETES_SERVICE_HOST"
+	yurthubHost       = "169.254.2.1"
+)
+
+// Register registers a filter
+func Register(filters *base.Filters) {
+	filters.Register(FilterName, func() (filter.ObjectFilter, error) {
+		return NewPodEnvFilter()
+	})
+}
+
+type podEnvFilter struct {
+	host string
+}
+
+func NewPodEnvFilter() (*podEnvFilter, error) {
+	return &podEnvFilter{}, nil
+}
+
+func (pef *podEnvFilter) Name() string {
+	return FilterName
+}
+
+func (pef *podEnvFilter) SupportedResourceAndVerbs() map[string]sets.Set[string] {
+	return map[string]sets.Set[string]{
+		"pods": sets.New("list", "watch", "get", "patch"),
+	}
+}
+
+func (pef *podEnvFilter) SetMasterServiceHost(host string) error {
+	pef.host = host
+	return nil
+}
+
+func (pef *podEnvFilter) SetMasterServicePort(portStr string) error {
+	// do nothing
+	return nil
+}
+
+func (pef *podEnvFilter) Filter(obj runtime.Object, _ <-chan struct{}) runtime.Object {
+	switch v := obj.(type) {
+	case *corev1.Pod:
+		return pef.mutatePodEnv(v)
+	default:
+		return v
+	}
+}
+
+func (pef *podEnvFilter) mutatePodEnv(req *corev1.Pod) *corev1.Pod {
+	proxyIP := pef.host
+	if proxyIP == "" {
+		proxyIP = yurthubHost
+	}
```

Comment:
It's not needed to verify `pef.host` whether it is an empty string or not in Filter function. we can use `pef.host` directly without verification.

Reply:
fair enough, I've removed the verification in a rebase